src/assets/VueOptionVsCompositionAPI/option-vs-composition-api.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Option Vs Composition API in Vue JS
+authorName: Sagar Gurung
+authorAvatar: https://avatars.githubusercontent.com/u/46086950?v=4
+authorLink: https://github.com/SagarGi
+createdAt: Dec 13, 2023
+tags: vue, option API, composition API, framework, frontend
+banner: https://blog.jankaritech.com/src/assets/VueOptionVsCompositionAPI/images/OptionVsCompositionAPI.png
+---
+
+## Background
+When it comes to defining or creating a component in the `Vue.js`, we have heard about the `Option API`. We can create specific aspects or behaviors of a component and how we want to build through it.
+It provides different properties or options such as `data`, `methods`, `computed`, and `watch` through which we can organize the behavior of a component systematically. But since `Vue 3`, we have
+another API called `Composition API` through which we can achieve the same thing. The `Composition API` is a new highlight for many vue developers but both approaches to define a component have their
+unique pros and cons. So debating which is best among these two would be a difficult one.
+
+In this blog, we will discuss how we can convert a component defined in `Option API` to `Composition API`. Also at the end, we will know how we can define the component with `Composition API`.
```

Comment:
The verb 'discuss' means to look at the advantages and disadvantages of options which you briefly do at the end of the blog post where as for the first part 'learn' or 'demonstrate' is maybe a more appropriate verb. Although the larger part of the blog is kind of a tutorial, it is good to mention the discussion part in the introduction, for example:

In this blog, we will learn how we can convert a component defined in Option API to Composition API and we will also discuss the advantages of of both APIs. At the end, we will know how we can define the component with Composition API.